outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -106,50 +107,64 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
 
     }
 
-    private getCompanyLayout= () => {
-        const company = this.state.company as Company;
-
-        if (company.isEmpty())
+    private updateCompanyRequest = () => {
+        if (!this.props.partner.email)
         {
-            let enrichAndCreate = null;
-            if (this.props.partner.isAddedToDatabase() && !this.enriched)
+            let enrichmentInfo = new EnrichmentInfo(EnrichmentInfoType.EnrichContactWithNoEmail,
+                _t("This contact has no email address, no company could be enriched."));
+            this.context.showTopBarMessage(enrichmentInfo);
+            return;
+        }
+
+        this.setState({isLoading: true});
+
+
+        const enrichRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.enrichAndUpdateCompany,
+            ContentType.Json, this.context.getConnectionToken(), {
+                partner_id: this.props.partner.id,
+            }, true);
+
+        this.context.addRequestCanceller(enrichRequest.cancel);
+
+        enrichRequest.promise.then((response) => {
+            const parsed = JSON.parse(response);
```

Comment:
"parsed" -> "enrichValues" or something more meaningful ?

Reply:
parsed also contain other entries like the json RPC version and not only the enrich values, so IMO renaming it to enrichValues can be confusing.

However, I think that if we call it parsed_result that would be better, but for consistency I think that we should also do the same for all other request methods.

Can you tell me what do you think about it?

Reply:
Generally speaking, it's not a good idea to base the name of the variable on the encoding / decoding you're applying.

It's like naming something you decode in base64 "base_64_variable", it does not tell you muchabout what it is.
A better option is to name is based on what it contains.

Here, if it contains more than just enrich values, then maybe something more global like "enrichResponse" ?

Reply:
Yeah true, variable name changed to enrichResponse.